### PR TITLE
feat: config

### PR DESCRIPTION
**Pull request by GoldenSheep402**
## Summary by CodeRabbit

- **New Features**
  - Enhanced configuration management with dynamic loading of module configurations, improving flexibility during initialization.
  - Improved logging for module configuration loading, enabling better tracking of potential issues.
  - Introduced a new method for generating YAML configuration files, allowing for better file management and error handling.
  - Added a structured configuration approach for modules, facilitating easier access to module settings.

- **Bug Fixes**
  - Improved error handling for configuration file generation and module loading processes to prevent crashes and provide feedback.

- **Documentation**
  - Updated comments in the code to clarify module registration and configuration management processes for easier understanding.

### CODE REVIEW

File: core/kernel/module.go [hunk 1]
```diff
@@ -24,7 +25,7 @@ type Module interface {
 	mustEmbedUnimplementedModule()
 }
 
-func (e *Engine) RegMod(mods ...Module) {
+func (e *Engine) RegMod(mods []Module) {
```
Comment (asjdf) [inline core/kernel/module.go]: 这里改数组是啥操作？

File: core/kernel/kernel.go [hunk 2]
```diff
@@ -44,6 +46,20 @@ func (e *Engine) StartModule() error {
 	hub := Hub{
 		Injector: e.Injector,
 	}
+	for _, module := range e.modules {
+		//if reflect.TypeOf(module.Config()) != nil &&
```
Comment (asjdf) [inline core/kernel/kernel.go]: 感觉注释可以移除

File: conf/config.go [hunk 3]
```diff
@@ -61,6 +61,24 @@ func GenYamlConfig(path string, force bool) error {
...
+//			err := os.WriteFile(path, data, 0644)
+//			if err != nil {
+//				return errors.New("Generate file with error: " + err.Error())
+//			}
+//			fmt.Println("Config file `config.yaml` generate success in " + path)
+//		} else {
+//			return errors.New(path + " already exist, use -f to Force coverage")
+//		}
+//		return nil
+//	}
```
[the remark this answers is not in the document]
Comment (asjdf) [reply conf/config.go]: 这个确实可以扬，主打一个简练

File: cmd/server/server.go [hunk 4]
```diff
@@ -25,14 +20,14 @@ import (
 var log = logx.NameSpace("cmd.server")
 
 var (
-	configYml string
-	StartCmd  = &cobra.Command{
+	ConfigPath string
```
Comment (asjdf) [inline cmd/server/server.go]: 这个export的目的是啥？cmd的设置不用export吧？

File: cmd/server/modList/list.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package modList
+
+import (
+	"github.com/juanjiTech/jframe/core/kernel"
+	"github.com/juanjiTech/jframe/mod/b2x"
+	"github.com/juanjiTech/jframe/mod/myDB"
+)
+
+var ModList = []kernel.Module{
+	&b2x.Mod{},
```
Comment (asjdf) [inline cmd/server/modList/list.go:14]: 调整一下默认启用的模块，参照之前的默认启用

**Review by asjdf — CHANGES_REQUESTED** [review]

File: core/kernel/module.go (hunk 1 above)
Comment (GoldenSheep402) [reply core/kernel/module.go]: 这样modList可以直接传进去

File: cmd/server/server.go (hunk 4 above)
Comment (GoldenSheep402) [reply cmd/server/server.go]: 草 忘删了

File: cmd/server/server.go [hunk 6]
```diff
@@ -25,14 +20,14 @@ import (
 var log = logx.NameSpace("cmd.server")
 
 var (
-	configYml string
-	StartCmd  = &cobra.Command{
+	configPath string
+	StartCmd   = &cobra.Command{
```
Comment (asjdf) [inline cmd/server/server.go:24]: lint

File: core/kernel/module.go (hunk 1 above)
Comment (asjdf) [reply core/kernel/module.go]: 这玩意是动态入参，你写三个点不就好了。“...”

**Review by asjdf — CHANGES_REQUESTED** [review]

**Comment by GoldenSheep402** [comment]
还真可以这么玩我去

**Review by asjdf — APPROVED** [review]

**Pull request merged**